### ISSUE TITLE
EventHub Namespace KeyVaultProperties is now required

### ISSUE THREAD

**Issue by aangelisc** [pulumi/pulumi-azure-nextgen#243]
It seems that the recent update (0.5.0) has introduced a change that has led to the `KeyVaultProperties` argument having to be defined in order to deploy a Namespace resource using the `v20180101preview` API. This issue did not exist in 0.4.0. 

The code below successfully deployed on 0.4.0:

```
const eventhubNamespace = new azureng.eventhub.v20180101preview.Namespace('name', {
  clusterArmId: eventHubCluster.id,
  resourceGroupName,
  location,
  namespaceName: 'name',
});
```

However, using 0.5.0 it leads to the following error:
```
Code="BadRequest" Message="All KeyVaultProperties must be specified as part of the Encryption object.
```

**Comment by mikhailshilkov** [comment]
Thank you for reporting.

The problem seems to be rooted in the fact that the `encryption` property is marked to be "flattened" [here](https://github.com/Azure/azure-rest-api-specs/blob/03af722477a8ae191de0225cc64fb0acfcf989da/specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/namespaces-preview.json#L745) which makes us promote its sub-properties to the top-level. 0.5.0 also applies default values in this case, which leads to the error.

It doesn't make sense to mark this property as flattened to me... I'll try to investigate why they did it and how Azure's own SDKs deal with that.

**Comment by mikhailshilkov** [comment]
Opened https://github.com/Azure/azure-rest-api-specs/issues/12648 upstream

**Comment by mikhailshilkov** [comment]
Okay, the specs look wrong but I don't expect a quick fix there. Meanwhile, I have a fix for this on our side, we'll ship it in 0.6.0. Sorry for the inconvenience.

**Comment by aangelisc** [comment]
No need to apologise, that works for me and I'm glad I could help. I'm aware of the pace with which the specs are updated given the issue I opened still hasn't been reviewed.
